google_shopping.js

```diff
@@ -2,4 +2,103 @@ var data = require("./products.json")
 
 // Write your solutions below
 
-console.log(data["items"]);
\ No newline at end of file
+//console.log(data["items"][0]['kind']);
+
+//QUESTION 1--------------------------
+
+
+var shoppingProducts = [];
+
+for (var i = 0; i < (data['items'].length); i++) {
+	if (data['items'][i]['kind'] === "shopping#product") {
+		shoppingProducts.push("Item " + i + " is a shopping#product.");
+	}
+}
+
+console.log(shoppingProducts);
+console.log(shoppingProducts.length + " items total."); 
+
+
+//Answer = 25, same as currentItemCount
+
+//QUESTION 2----------------------------
+
+
+var backOrdered = [];
+
+for (var i = 0; i < (data['items'].length); i++) {
+	for (var j = 0; j < data['items'][i]['product']['inventories'].length; j++) {
+		if ((data['items'][i]['product']['inventories'][j]['availability']) === "backorder") {
+			backOrdered.push("Item " + i + " has a backorder.");
+		}
+	}
+}
+
```

Comment:
I don't think you need a nested for loop here, the following code will work just as well:

```for (var i = 0; i < data["items"].length; i++) {```
	```if (data["items"][i].product.inventories[0].availability == 'backorder') {```
	```backOrdered.push("Item " + i + " has a backorder.");```
	```};```
```};```